### ISSUE TITLE
Implement Parallelization Factor for Kinesis/DynamoDB Streams consumers

### ISSUE THREAD

**Issue by aaleksandrov** [terraform-providers/terraform-provider-aws#11214]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Recently AWS added support for `parallelization factor` parameter for Kinesis and DynamoDB event consumers https://aws.amazon.com/blogs/compute/new-aws-lambda-scaling-controls-for-kinesis-and-dynamodb-event-sources/
Please support it

### New or Affected Resource(s)

aws_lambda_event_source_mapping

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by bflad** [comment]
Open pull request for this functionality can be found here: https://github.com/terraform-providers/terraform-provider-aws/pull/11100

**Comment by bflad** [comment]
Related request for this functionality: https://github.com/terraform-providers/terraform-provider-aws/issues/11386

**Comment by bflad** [comment]
Support for the additional `aws_lambda_event_source_mapping` arguments has been merged and will release with version 2.51.0 of the Terraform AWS Provider, end of next week. Thanks to @kostas-theo for the implementation. 👍
